modules/social_features/social_user_export/src/Plugin/UserExportPluginBase.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace Drupal\social_user_export\Plugin;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Datetime\DateFormatterInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\profile\Entity\ProfileInterface;
+use Drupal\taxonomy\Entity\Term;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Base class for User export plugin plugins.
+ */
+abstract class UserExportPluginBase extends PluginBase implements UserExportPluginInterface, ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  public $entityTypeManager;
+
+  /**
+   * The date formatter.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  public $dateFormatter;
+
+  /**
+   * The database.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  public $database;
+
+  /**
+   * UserExportPluginBase constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter.
+   * @param \Drupal\Core\Database\Connection $database
+   *   The database connection.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, DateFormatterInterface $date_formatter, Connection $database) {
```

Comment:
I'm not sure if there's an easy way around this, but we're adding services like `database` and `dateFormatter` that we do not user for every export plugin. It might be more clean to add those in the plugins that actually use them.

Reply:
I don't think we should do that. That will make things even moe messy. I also have a feeling that the impact on performance will not be that big anyways. And especially entity type manager and database are very valuable services and you will use the most likely.